chat-widget/automation_tests/configuration/CustomEnvironment.js

```diff
@@ -0,0 +1,39 @@
+/* eslint-disable @typescript-eslint/no-var-requires */
+const PlaywrightEnvironment =
+  require("jest-playwright-preset/lib/PlaywrightEnvironment").default;
+/* eslint-disable @typescript-eslint/no-var-requires */
+const fs = require("fs");
+const testDoneEventName = "test_done";
+class CustomEnvironment extends PlaywrightEnvironment {
+    async handleTestEvent(event) {
+        if (event.name === testDoneEventName && event.test.errors.length > 0) {
+            try {
+                const parentName = event.test.parent.name.replace(/\W/g, "_");
+                const specName = event.test.name.replace(/\W/g, "_");
+                const contexts = this.global.browser.contexts();
+                for (let [contextIndex, currentContext] of contexts.entries()) {
+                    const contextPages = currentContext.pages();
+                    for (let [pageIndex, currentPage] of contextPages.entries()) {
+                        if (currentPage) {
+                            try {
+                                await currentPage.screenshot({
```

Comment:
Why are we taking screenshot here?

Reply:
it is used to capture screenshot if there is an error in testcase
as we are using playwright with jest then this is required.

I will check if we can use Playwright Test runner instead of Jest runner.